pkg/channel/lookup.go

```diff
@@ -328,3 +329,18 @@ func NewMoreThanOneTemplateCandidateErr(component v1beta2.Module,
 	return fmt.Errorf("%w: more than one module template found for module: %s, candidates: %v",
 		ErrTemplateNotIdentified, component.Name, candidates)
 }
+
+func saveDescriptorToCache(template ModuleTemplateTO) {
+	if template.ModuleTemplate != nil {
```

Comment:
Write this condition with `if template.ModuleTemplate == nil`, then you can return error immediately and have less nested if block.